From 04e315b0f826486e4cb213e29ea7b4411bced739 Mon Sep 17 00:00:00 2001
From: Ciprian Costea <ciprianmarian.costea@nxp.com>
Date: Mon, 29 Jul 2024 14:35:20 +0300
Subject: [PATCH] sdhci-esdhc-imx: Enable CQE support
Content-Type: text/plain; charset=UTF-8

Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
---
 drivers/mmc/host/sdhci-esdhc-imx.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/mmc/host/sdhci-esdhc-imx.c b/drivers/mmc/host/sdhci-esdhc-imx.c
index 47fbaf505ac6..f34e81e46505 100644
--- a/drivers/mmc/host/sdhci-esdhc-imx.c
+++ b/drivers/mmc/host/sdhci-esdhc-imx.c
@@ -331,6 +331,7 @@ static struct esdhc_soc_data usdhc_s32cc_data = {
 			| ESDHC_FLAG_HS400
 			| ESDHC_FLAG_HS400_ES
 			| ESDHC_FLAG_MAN_TUNING
+			| ESDHC_FLAG_CQHCI
 			| ESDHC_FLAG_BROKEN_SDR104,
 };
 
-- 
2.45.2

